game-backend/src/game-engine/mechanics/createSpawner.ts

```diff
@@ -0,0 +1,27 @@
+import { State, GameObject, Position } from '@darwin/types';
+import { generateFreePosition } from '../../helper/fields';
+import produce from '../../helper/produce';
+
+type params = {
+  limit: (state: State) => number;
+  count: (state: State) => number;
+  spawn: (position: Position) => GameObject;
+};
+
+const createSpawner = ({ limit, count, spawn }: params) => (
```

Comment:
Now that it's extracted, there should be a unit test for `createSpawner.ts` but currently it's tested twice using the separate tests. What's the benefit of testing it for each usage again?